modules/sorts/test/test.cpp

```diff
@@ -0,0 +1,12 @@
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE merge_sort_test
+#include <boost/test/unit_test.hpp>
+#include "merge_sort.h" // declares type foo, header is in the core module
+#include <vector>
+
+BOOST_AUTO_TEST_CASE(TestName) {
```

Comment:
You should also use a descriptive test case name. If you want to see how these affect how the tests are displayed take a look at the [tests from the computer-vision program](http://bmwinger.linuxstart.ca/jenkins/job/Computer_Vision/21/testReport/).
Test results in Jenkins are grouped first by module (as specified with BOOST_TEST_MODULE), then by file (I think using the BOOST_AUTO_TEST_SUITE macro will overrides this, haven't tested that though) and then by name.

With this in mind, merge_sort_test would be a better name for the suite than for the module, considering that it is part of the sorts module in the code (see [here](http://www.boost.org/doc/libs/1_61_0/libs/test/doc/html/boost_test/tests_organization/test_suite.html) if you want to try and see how jenkins will display tests when you make use of BOOST_AUTO_TEST_SUITE). 

While it's not a huge issue, I think I should also point out that calling the file test.cpp (which admittedly is actually the name of the first test file added to the computer-vision repo) isn't terribly descriptive either considering there might potentially be other test files in the sorts module.